### PR TITLE
Policy.json should be defined by user

### PR DESCRIPTION
**Pull request by dukov**
Implementation has been moved from template based to pillar based. User
can override and add values to policy.json by creating flat key-value
structure under keystone:server:policy.

Change-Id: Iebfef4cbef2eb531721c0b604560464b12508e3d

### CODE REVIEW

**Comment by epcim** [comment]
+1

File: keystone/server.sls [hunk 1]
```diff
@@ -103,9 +103,13 @@ keystone_group:
     - service: keystone_service
   {%- endif %}
 
+{% import_json "/etc/keystone/policy.json" as default_policy %}
```
Comment (fpytloun) [inline keystone/server.sls]: Good idea to use file.serialize but this part probably won't work on clean installation where keystone is not yet installed.

**Comment by epcim** [comment]
@dukov do you have any update or rewrite? An ETA?

**Comment by dukov** [comment]
Will try to propose new approach this week

File: keystone/server.sls [hunk 2]
```diff
@@ -104,8 +104,10 @@ keystone_group:
   {%- endif %}
 
 /etc/keystone/policy.json:
-  file.managed:
-  - source: salt://keystone/files/{{ server.version }}/policy-v{{ server.api_version }}.json
```
Comment (fpytloun) [inline keystone/server.sls:108]: You should also remove old files that won't be used anymore.

Comment (fpytloun) [reply keystone/server.sls:108]: And mention this feature in README with example

File: _states/keystone_policy.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python
```
Comment (fpytloun) [inline _states/keystone_policy.py]: ```#!/usr/bin/env python```

File: _states/keystone_policy.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+# Copyright 2017 Mirantis, Inc.
```
Comment (fpytloun) [inline _states/keystone_policy.py]: It's better to use `Mirantis, Inc. et al.` to include other contributors. Or don't add the license at all as it's already in root of this repository.

File: keystone/server.sls (hunk 2 above)
Comment (dukov) [reply keystone/server.sls:108]: fixed

File: _states/keystone_policy.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    :param name:          Name of the resource
+    :param override_data: User defined data with overrides
+    '''
+    with open(JSON_LOCATION) as policy_json:
+        json_content = json.load(policy_json)
+
+    _deep_merge(json_content, override_data)
+
+    kwargs['dataset'] = json_content
+    ret = __states__['file.serialize']('/etc/keystone/policy.json', **kwargs)
```
Comment (epcim) [inline _states/keystone_policy.py:52]: use JSON_LOCATION

File: _states/keystone_policy.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        override_key: override_value
+    - formatter: json
+
+'''
+import logging
+import json
+
+log = logging.getLogger(__name__)
+
+JSON_LOCATION = '/etc/keystone/policy.json'
```
Comment (epcim) [inline _states/keystone_policy.py:23]: JSON_LOCATION should be configurable attribute

File: _states/keystone_policy.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+.. code-block:: yaml
+
+/etc/keystone/policy.json:
+  keystone_policy.present:
+    - override_data:
+        override_key: override_value
+    - formatter: json
+
+'''
+import logging
```
Comment (epcim) [inline _states/keystone_policy.py:18]: pls double check these are standard library, otherwise we should install python packages first.

**Comment by epcim** [comment]
@dukov was merged correct version or you have newer on gerrit? We may revert for easier merge if you have reworked code elsewhere.

Also, ples see me comments I posted ext-post.

**Comment by dukov** [comment]
@epcim I'd prefer to have this work here to make these changes publically available as PR

**Comment by epcim** [comment]
sure, for public repos I preffer Gh as well

**Pull request merged**